processor/src/trace.rs

```diff
@@ -21,21 +26,41 @@ impl ExecutionTrace {
             system,
             decoder: _,
             stack,
-            hasher: _,
-            bitwise: _,
-            memory: _,
+            hasher,
+            bitwise,
+            memory,
             advice: _,
         } = process;
 
+        // get the length required to hold all execution trace steps
+        let aux_trace_len = hasher.trace_len() + bitwise.trace_len() + memory.trace_len();
+        let mut trace_len = usize::max(stack.trace_len(), aux_trace_len);
+        // pad the trace length to the next power of 2
+        if !trace_len.is_power_of_two() {
+            trace_len = trace_len.next_power_of_two();
+        }
+
+        // allocate columns for the trace of the auxiliary table
+        let mut aux_table_trace: AuxiliaryTableTrace = (0..AUXILIARY_TABLE_WIDTH)
+            .map(|_| Vec::<Felt>::with_capacity(trace_len))
```

Comment:
I wonder if here we could allocate initialized memory right away (e.g., via `Felt::zeroed_vector()`). This way, we would need to write only non-zero values into the table.

Reply:
Yeah, I thought about initializing it immediately. The reason I didn't was so we could use resize to set the selectors rather than iterating to update them, which I thought was your preference, although I could have misunderstood. There aren't that many selector values in comparison to the number of places that need to be zeroed out, especially since initialization is required before filling the traces.

One other thing that's not fully clear to me (indicating I just need more reading & to run some benchmarks), is whether the approach you suggested gets the speed benefits of vec![0; n]. I know that vec! is by far the fastest option for initializing with all zeroes, but I wasn't sure if it's initialized using the same approach once it's in a struct or if it repeatedly clones values, more like vec![1; n], in which case my understanding is that the speed difference between vec! and allocating the capacity then resizing isn't as significant, and so we might not want the extra iteration to update values.

Reply:
I've added an additional comment in the code for now